### ISSUE TITLE
How to add many characteritics labels (SegmentedPropertyTypeModifiers?) to a SEG? 

### ISSUE THREAD

**Issue by gacou54** [QIICR/QuantitativeReporting#277]
Hi,

We evaluate whether it would be possible to associate several characteristics with a pulmonary nodule segmented with the Quantitative Reporting extension/workflow, in particular some characteristics identified in Table 1 of https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3056962/ (It uses the RADLEX terminologies).

I am wondering if it is possible to input the nodule characteristics in the SEG DICOM (likely somewhere in the `SegmentSequence`) with Quantitative Reporting, and if yes how to do it? 

I identified `SegmentedPropertyTypeModifierCodeSequence` https://github.com/QIICR/QuantitativeReporting/blob/166d695ea2965c1a40676107931a57aa2cb6cfad/DICOMPlugins/DICOMSegmentationPlugin.py#L714

that _may_ be OK to use for this, but unfortunately I think the Terminologies Module from Slicer only allows one TypeModifier (it's a select widget).

TLDR: Is there a way to add more than one characteristic label (from a predetermined terminologies list) to a Segmenation within Quantitative Reporting workflow?

Thanks in advance :)

**Comment by fedorov** [comment]
@gacou54 if you want to describe the characteristics of a lung nodule, you would not use DICOM SEG, but DICOM SR for this.

Given your reference, you might want to take a look at this manuscript that applies the above in practice:

> Fedorov, A., Hancock, M., Clunie, D., Brochhausen, M., Bona, J., Kirby, J., Freymann, J., Pieper, S., J W L Aerts, H., Kikinis, R. & Prior, F. DICOM re-encoding of volumetrically annotated Lung Imaging Database Consortium (LIDC) nodules. Med. Phys. (2020). http://dx.doi.org/10.1002/mp.14445

You can access the data referenced in that paper from Imaging Data Commons: https://portal.imaging.datacommons.cancer.gov/explore/filters/?collection_id=lidc_idri.

I am traveling the coming week, but let me know if you have questions and I will try to help when I am back.
​

**Comment by gacou54** [comment]
Thanks for the response and the (very interesting) link. Indeed the SRs seem more appropriate for the characteristics. We will most likely take this route.

Thanks :)
(You can close the issue)

**Comment by fedorov** [comment]
Excellent! Take a look at dcmqi and the accompanying code for lidc conversion!

**Comment by Just5D** [comment]
Is there a way to display the lung nodule attribute score from SR in the QR extension, and then edit, modify, and save it?

**Comment by fedorov** [comment]
@Just5D no, this is not possible unfortunately. DICOM SR has the capabilities to encode such a score, but QR does not have the interface to support this, and I do not have the resources to implement it.
